### PR TITLE
Release 1.32.0-rc.2

### PR DESCRIPTION
**Pull request by shizuchanw**
**What type of PR is this?**
> Uncomment only one, leave it on its own line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Bump k8s.io depedencies to v1.32.0-rc.2 and golang version to 1.23 to start releasing 1.32.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module k8s.io/cloud-provider-aws
 
-go 1.22.7
+go 1.23.0
```
Comment (kmala) [inline go.mod:3]: you have to update other places also for go version like https://github.com/kubernetes/cloud-provider-aws/blob/master/Dockerfile#L17

Comment (shizuchanw) [reply go.mod:3]: Updated go version

**Comment by shizuchanw** [comment]
```
go mod edit -replace k8s.io/api=../../k8s.io/kubernetes/staging/src/k8s.io/api
go: downloading go1.23.4 (linux/amd64)
go: download go1.23.4: golang.org/toolchain@v0.0.1-go1.23.4.linux-amd64: verifying module: checksum database disabled by GOSUMDB=off
make: *** [Makefile:171: switch-to-latest-k8s] Error 1
```
Looks like the test failed when downloading go1.23.4

**Pull request merged**